Question 2 :

(a) Researcher A thinks that child height depends on father height, mother height or gender

Similar to example, we assume negation is correct and see if our data is consistent with that or not.

null hypothesis, H0 would be None of father height, mother height, or gender affects child height, in this case the coefficients of father height, mother height and gender will be zero.

H1 would be atleast one of these would be non-zero.


In [ ]:
install.packages("HistData")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
library(HistData)
data(GaltonFamilies)

GaltonFamilies$gender <- factor(GaltonFamilies$gender)
GaltonFamilies$gender <- relevel(GaltonFamilies$gender, ref = "female")

lmod <- lm(childHeight ~ father + mother + gender, data = GaltonFamilies)

summary(lmod)


Call:
lm(formula = childHeight ~ father + mother + gender, data = GaltonFamilies)

Residuals:
    Min      1Q  Median      3Q     Max 
-9.5247 -1.4653  0.0943  1.4860  9.1201 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 16.52124    2.72720   6.058    2e-09 ***
father       0.39284    0.02868  13.699   <2e-16 ***
mother       0.31761    0.03100  10.245   <2e-16 ***
gendermale   5.21499    0.14181  36.775   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.165 on 930 degrees of freedom
Multiple R-squared:  0.6354,	Adjusted R-squared:  0.6342 
F-statistic: 540.3 on 3 and 930 DF,  p-value: < 2.2e-16


p value for all (fatherr, mother and gender) are significantly smaller than 0.005 (our alpha), so we reject the null hypothesis. Father's height, mother's height and gender have impact on the child height. For how much affect we can look at the estimate for the impact, for 1 increase in father's height child height increases by 0.39 and for mother's 0.31 keeping others constant. Males are 5.21 taller than females.

(b)
Researcher B thinks that, given father and mother heights, boys and girls are expected to have different height.
Null hypothesis (H0), no effect of gender on height given mother and father height, meaning coefficient of gender will be zero.
H1, coefficient is non-zero, gender affects height.


In [ ]:
pvalue_gender <- coef(summary(lmod))["gendermale", "Pr(>|t|)"]
pvalue_gender


[1] 1.678202e-183

p-value for gender is less than alpha (0.005) so our null hypothesis is incorrect and thus gender has effect on height of children given mother and father height, so boys and girls have different height.

(c) Researcher C thinks that, given father and mother heights, boys on average are more than 4 inches taller than girls.

Null hypothesis (H0) is that height of boys is on average 4 inches less than or equal to girls, means the coefficient of gender will be less than or equal to 4

H1, coefficient of gender will be greater than 4

In [ ]:
betahat3 <- coef(lmod)["gendermale"]
se.betahat3 <- coef(summary(lmod))["gendermale", "Std. Error"]

t3obs <- (betahat3 - 4) / se.betahat3
pvalue <- 1 - pt(t3obs, df = nrow(GaltonFamilies) - length(coef(lmod)))
pvalue

gendermale 
         0

The p value is so close to zero, obviously less than 0.005 hence Null hypothesis is incorrect, so boys on average are more than 4 inches taller than girls.

(d) Researcher D thinks that child height is associated with at least one of father height and mother height.

Null hypothesis (H0), child height is not associated with either mother or father (total - union of both), here both coefficeients of mother and father height will be zero.

H1, Atleast one of the coefficients of mother, father height is not zero.

In [ ]:
pvalue_father <- coef(summary(lmod))["father", "Pr(>|t|)"]
pvalue_mother <- coef(summary(lmod))["mother", "Pr(>|t|)"]

pvalue_father
pvalue_mother

[1] 4.818605059947556888656e-39

[1] 2.061353885231034041538e-23

both p values are less than 0.005, we reject the null hypothesis, hence child height is associated with at least one of mother and father height.

(e) Researcher E believes that child height has a relationship with father and mother heights and does not believe that boys, on average, are 3 inches taller than girls given father and mother heights.

We are testing if boys are 3 inches taller or not effectively.

Null hypothesis (H0), boys are three inches taller than girls, coefficient of boys height is 3.

H1, beta3 not equals to 3

In [ ]:
betahat3 <- coef(lmod)["gendermale"]
se.betahat3 <- coef(summary(lmod))["gendermale", "Std. Error"]

t3obs <- (betahat3 - 3) / se.betahat3
pvalue <- 2 * (1 - pt(abs(t3obs), df = nrow(GaltonFamilies) - length(coef(lmod))))
pvalue


gendermale 
         0

just like for part (c), here p value is very close to zero. so we can conclude our null hypothesis is incorrect. So, boys are not on average 3 inches taller than girls.

Question 3

In [ ]:
conf <- confint(lmod, level = 0.95)

b3_conf <- conf["gendermale", ]
b3_conf

2.5 %                  97.5 % 
4.936686730241474307945 5.493291971621241387425

Researcher B thinks that, given father and mother heights, boys and girls are expected to have different height.
Null hypothesis (H0), no effect of gender on height given mother and father height, meaning coefficient of gender will be zero.
H1, coefficient is non-zero, gender affects height.

Here, the confidence interval doesn't include zero, so the coefficient of gender has impact, hence our null hypothesis is incorrect. Reasearcher B is correct to think that, given father and mother heights, boys and girls are expected to have different heights.

Calculation for probability from question 4

In [ ]:
t_value <- 4.683
df <- 28

p_value <- 2 * (1 - pt(t_value, df))
p_value

[1] 6.596323374119528182291e-05